research/src/fetch_campaign_donations.py

```diff
@@ -0,0 +1,346 @@
+import pandas as pd
+import urllib.request
+import os        
+import sys
+import zipfile
+import time
+
+def reporthook(blocknum, blocksize, totalsize):
+    readsofar = blocknum * blocksize
+    if totalsize > 0:
+        percent = readsofar * 1e2 / totalsize
+        s = "\r%5.1f%% %*d / %d" % (
+            percent, len(str(totalsize)), readsofar, totalsize)
+        sys.stderr.write(s)
+        if readsofar >= totalsize: # near the end
+            sys.stderr.write("\n")
+    else: # total size is unknown
+        sys.stderr.write("read %d\n" % (readsofar,))
+        
+url = 'http://agencia.tse.jus.br/estatistica/sead/odsele/prestacao_contas/prestacao_contas_2010.zip'
+file_name = 'prestacao_contas_2010.zip'
+print("Downloading 2010 donation data...")
+urllib.request.urlretrieve(url, file_name, reporthook)
+
+print("Uncompressing downloaded data.")
+with zipfile.ZipFile('prestacao_contas_2010.zip',"r") as zip_ref:
+    zip_ref.extractall("prestacao_contas_2010")
+os.remove("prestacao_contas_2010.zip")
```

Comment:
You basically repeat this block of code for every year. When you have things like that it's better to isolate this logic as a function and pass only what really changes from year to year.